torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -763,23 +763,24 @@ static void fuseListConstructListUnpack(Block* b) {
 
 // https://github.com/pytorch/pytorch/wiki/PyTorch-ONNX-exporter#quantized-model-export
 static void eraseTupleConstruct(Block* block) {
-  size_t index = 0;
+  std::vector<Value*> new_block_outputs;
   // TupleConstruct is generated from the symbolics in quantized domain, and
   // consumed by other quantized operators. The remained TupleConstruct should
   // be at the output of the blocks.
   for (auto* output : block->outputs()) {
     auto output_node = output->node();
     if (output_node->kind() == prim::TupleConstruct) {
-      block->eraseOutput(index);
-      size_t input_index = 0;
       for (auto* input : output_node->inputs()) {
-        block->insertOutput(index + (input_index++), input);
+        new_block_outputs.emplace_back(input);
       }
-      index += input_index;
     } else {
-      index++;
+      new_block_outputs.emplace_back(output);
     }
   }
+  block->removeAllOutputs();
```

Comment:
not sure if this is a premature optimization, but maybe keep a bool like:
`found_tuple_construct` and only remove and register new outputs if it's true?

Up to you if you think this is a good idea.

Reply:
updated.